### ISSUE TITLE
'conda update -all' with mkl 11.3.1-0 and numpy 1.10.4

### ISSUE THREAD

**Issue by wikiped** [conda/conda#2034]
After updating `numpy` to `1.10.4` `mkl` is getting installed automatically (if you let it of course). But after installing `mkl` (latest version `11.3.1-0`) running `conda update --all` offers to downgrade all `numpy` dependant packages:

    The following packages will be downloaded:

    package                    |            build
    ---------------------------|-----------------
    numpy-1.9.3                |          py27_p0         2.8 MB  defaults
    bottleneck-1.0.0           |       np19py27_0         142 KB  defaults
    h5py-2.5.0                 |       np19py27_2         1.4 MB  defaults
    numexpr-2.4.4              |      np19py27_p0         137 KB  defaults
    scipy-0.16.0               |      np19py27_p0        83.7 MB  defaults
    matplotlib-1.4.3           |       np19py27_3        40.8 MB  defaults
    numba-0.23.1               |       np19py27_0         1.5 MB  defaults
    pandas-0.17.0              |       np19py27_0         5.8 MB  defaults
    pymc-2.3.5                 |       np19py27_0         4.4 MB  defaults
    pysal-1.10.0               |       np19py27_0        11.1 MB  defaults
    pytables-3.2.2             |       np19py27_0         1.4 MB  defaults
    scikit-learn-0.16.1        |      np19py27_p0         3.2 MB  defaults
    basemap-1.0.7              |       np19py27_0       120.5 MB  defaults
    scikit-image-0.11.3        |       np19py27_0        15.8 MB  defaults
    statsmodels-0.6.1          |       np19py27_0         4.6 MB  defaults
    ------------------------------------------------------------
                                           Total:       297.5 MB

    The following packages will be UPDATED:

    basemap:      1.0.7-np110py27_0  defaults --> 1.0.7-np19py27_0   defaults
    bottleneck:   1.0.0-np110py27_0  defaults --> 1.0.0-np19py27_0   defaults
    numba:        0.23.1-np110py27_0 defaults --> 0.23.1-np19py27_0  defaults
    pysal:        1.10.0-np110py27_0 defaults --> 1.10.0-np19py27_0  defaults
    scikit-image: 0.11.3-np110py27_0 defaults --> 0.11.3-np19py27_0  defaults
    statsmodels:  0.6.1-np110py27_0  defaults --> 0.6.1-np19py27_0   defaults

    The following packages will be DOWNGRADED:

    h5py:         2.5.0-np110py27_4  defaults --> 2.5.0-np19py27_2   defaults
    matplotlib:   1.5.1-np110py27_0  defaults --> 1.4.3-np19py27_3   defaults
    numexpr:      2.4.6-np110py27_1  defaults --> 2.4.4-np19py27_p0  defaults [mkl]
    numpy:        1.10.4-py27_0      defaults --> 1.9.3-py27_p0      defaults [mkl]
    pandas:       0.17.1-np110py27_0 defaults --> 0.17.0-np19py27_0  defaults
    pymc:         2.3.6-np110py27_0  defaults --> 2.3.5-np19py27_0   defaults
    pytables:     3.2.2-np110py27_1  defaults --> 3.2.2-np19py27_0   defaults
    scikit-learn: 0.17-np110py27_2   defaults --> 0.16.1-np19py27_p0 defaults [mkl]
    scipy:        0.17.0-np110py27_0 defaults --> 0.16.0-np19py27_p0 defaults [mkl]

Since `mkl` package version has no obvious declaration of `numpy=1.9` dependance (like `np19py27_0`) this comes as a surprise and not an easy thing to pin-point when trying to figure out what drives this behavior (in case `mkl` was installed before numpy update to `1.10.4`).

So is `mkl` actually not compatible with `numpy 1.10.4`?

    Current conda install:

                 platform : win-64
            conda version : 3.19.1
      conda-build version : 1.19.0
           python version : 2.7.11.final.0
         requests version : 2.9.1
         root environment : D:\Anaconda  (writable)
      default environment : D:\Anaconda
         envs directories : D:\Anaconda\envs
            package cache : D:\Anaconda\pkgs
             channel URLs : https://repo.continuum.io/pkgs/free/win-64/
                            https://repo.continuum.io/pkgs/free/noarch/
                            https://repo.continuum.io/pkgs/pro/win-64/
                            https://repo.continuum.io/pkgs/pro/noarch/
              config file : D:\Anaconda\.condarc
        is foreign system : False

**Comment by aoeftiger** [comment]
I encountered the same problem.

**Comment by wikiped** [comment]
Thank you for the link. Unfortunately this is not helping in my situation (although I tried removing `mkl` folder manually after uninstalling `mkl`). I actually don't have problems with using `numpy` - it's just when you run `conda update --all` you get all those false downgrade warnings, making `update --all` option useless.

**Comment by wikiped** [comment]
Looks like this has been fixed with `conda >= 4.0`

**Comment by dprada** [comment]
@wikiped Thanks. 

Updating conda to 4.0 solved this issue and also the problem of libgfortran #2177 and https://github.com/ContinuumIO/anaconda-issues/issues/686.

I had the following message importing scipy.stats:
```
ImportError: libgfortran.so.1: cannot open shared object file: No such file or directory
```
and also:
```
ImportError: .../lib/python2.7/site-packages/scipy/special/../../../../libgfortran.so.3: version `GFORTRAN_1.4' not found (required by /usr/lib/liblapack.so.3gf)
```

In my case this was solved Updating conda to 4.0. After that numpy and scypi were also updated to:
```
    numexpr:      2.4.4-np110py27_p0  [mkl] --> 2.5-np110py27_0   
    numpy:        1.10.2-py27_p0      [mkl] --> 1.10.4-py27_1     
    pytables:     3.2.2-np110py27_0   --> 3.2.2-np110py27_1 
    scikit-learn: 0.17-np110py27_p1   [mkl] --> 0.17.1-np110py27_0
    scipy:        0.16.1-np110py27_p0 [mkl] --> 0.17.0-np110py27_2
```
